libwallet/src/api_impl/owner.rs

```diff
@@ -754,16 +778,51 @@ where
 {
 	let mut sl = slate.clone();
 	check_ttl(w, &sl)?;
-	let context = w.get_private_context(keychain_mask, sl.id.as_bytes())?;
+	let mut context = w.get_private_context(keychain_mask, sl.id.as_bytes())?;
+	let keychain = w.keychain(keychain_mask)?;
 	let parent_key_id = w.parent_key_id();
 
-	// since we're now actually inserting our inputs, pick an offset and adjust
-	// our contribution to the excess by offset amount
-	// TODO: Post HF3, this should allow for inputs to be picked at this stage
-	// as opposed to locking them prior to this stage, as the excess to this point
-	// will just be the change output
+	if let Some(args) = context.late_lock_args.take() {
+		// Transaction was late locked, select inputs+change now
+		// and insert into original context
+
+		let current_height = w.w2n_client().get_chain_tip()?.0;
+		let mut temp_sl =
+			tx::new_tx_slate(&mut *w, context.amount, false, 2, false, args.ttl_blocks)?;
+		let temp_context = selection::build_send_tx(
```

Comment:
I think there may be an issue here with `build_send_tx()` when late locking.
We take the amount from the slate, not from the local private ctx.

Reply:
We restore the original fee and amount on the slate based on out local ctx in `repopulate_tx()` but this happens _after_ we build the tx (based on the slate) during late locking finalization.

Reply:
Presumably some subset of data is "set once", transmitted via the slate, but should always be read from the local ctx by either party once set.
Ideally this data is not even included in the slate in subsequent communication.

Reply:
Quick summary of discussion - 

When we call `build_send_tx()` we pass in the _temporary_ slate `temp_sl`, not the slate passed from the recipient.
This temporary slate contains `amount` from our local `context`.

We then update `context` with input and output ids from the _temporary_ context `temp_context` created by `build_send_tx()`.
The updated private `context` is then persisted locally.